pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -186,16 +200,29 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 	}, lastMaintenances...)
 
 	if len(lastMaintenances) > maxTrackedMaintenanceOperations {
-		newShoot.Status.LastMaintenances = lastMaintenances[0:3]
+		shoot.Status.LastMaintenances = lastMaintenances[0:3]
 	} else {
-		newShoot.Status.LastMaintenances = lastMaintenances
+		shoot.Status.LastMaintenances = lastMaintenances
 	}
 
-	if err := r.Client.Status().Patch(ctx, newShoot, patch); err != nil {
+	// here `gardener.cloud/operation`: `maintain` is removed so when patch call triggers
+	// another reconciliation it should not be picked by the controller
+	if hasMaintainNowAnnotation(shoot) {
+		delete(shoot.Annotations, v1beta1constants.GardenerOperation)
+	}
+
+	if err := r.Client.Status().Patch(ctx, shoot, patch); err != nil {
```

Comment:
What if the patch status fails and the controller is restarted? Or even if the patch succeeds but the controller is restarted before the maintenance operation is completely finished? Then the maintenance operation will not be retried.

Reply:
In the first case if the patch call fails and the controller is restarted then the maintenance operation will be retried cause shoot will still have the required maintain operation annotation, but in the second case if the patch passes and controller restarts before the update call then maintenance operation will not be retried. I will open a revert PR for https://github.com/gardener/gardener/pull/6725 to not delay the release process.
/close